bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/form/TitleImpl.java

```diff
@@ -0,0 +1,139 @@
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.forms.core.components.internal.models.v1.form;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.*;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.forms.core.components.internal.Heading;
+import com.adobe.cq.forms.core.components.internal.form.FormConstants;
+import com.adobe.cq.forms.core.components.models.form.Title;
+import com.adobe.cq.forms.core.components.util.AbstractComponentImpl;
+import com.adobe.cq.forms.core.components.util.ComponentUtils;
+import com.adobe.cq.wcm.core.components.models.datalayer.ComponentData;
+import com.adobe.cq.wcm.core.components.models.datalayer.builder.DataLayerBuilder;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.i18n.I18n;
+import com.day.cq.wcm.api.WCMMode;
+import com.day.cq.wcm.api.designer.Style;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@Model(
+    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adapters = { Title.class, ComponentExporter.class },
+    resourceType = FormConstants.RT_FD_FORM_TITLE_V1)
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+
+public class TitleImpl extends AbstractComponentImpl implements Title {
+
+    public TitleImpl() {
+
+    }
+
+    /**
+     * The current resource.
+     */
+    @SlingObject
+    private Resource resource;
+
+    @ScriptVariable(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @JsonIgnore
+    @Nullable
+    private Style currentStyle;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Nullable
+    private String type;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL, name = JcrConstants.JCR_TITLE)
+    @Nullable
+    private String title;
+
+    private Heading heading;
+
+    /**
+     * Translation of the title property
+     */
+    private I18n i18n;
+
+    public TitleImpl(I18n i18n) {
+        this.i18n = i18n;
+    }
+
+    @PostConstruct
+    private void initModel() {
+        if (request != null && i18n == null) {
+            i18n = null;
+        }
+
+        if (heading == null) {
+            heading = Heading.getHeading(type);
+            if (heading == null && currentStyle != null) {
+                heading = Heading.getHeading(currentStyle.get(PN_DESIGN_DEFAULT_TYPE, String.class));
```

Comment:
unit test for this pls.

Reply:
I had already written a test case for it, named testGetTitleResourcePageStyleType(). Not sure what you mean here. Can you please elaborate?